samples/cellular/modem_shell/src/fota/fota_shell.c

```diff
@@ -33,8 +33,7 @@ static int cmd_fota_download(const struct shell *shell, size_t argc,
 	} else if (strcmp(argv[1], "au") == 0) {
 		fota_server = fota_server_au;
 	} else {
-		mosh_error("FOTA: Unknown server: %s", argv[1]);
-		return -EINVAL;
+		fota_server = argv[1];
```

Comment:
I tried to download another file which is not a delta update. The first attempt ends up in an error and an "unexpected state transition":
```
mosh:~$ fota download http://test-eu.nrfcloud.com/small_1024.txt foo
FOTA: Starting download...
FOTA: Download started
[00:03:52.519,042] <inf> downloader: Setting up TLS credentials, sec tag count 1
[00:03:52.519,195] <inf> downloader: Connecting to 13.48.128.56
FOTA: Error, download failed
[00:04:16.022,705] <err> downloader: Unable to connect, errno 116
[00:04:16.022,979] <err> downloader: Failed to set http recv timeout, err -22
[00:04:16.023,010] <err> fota_download: Download client error event -22
[00:04:16.027,984] <wrn> downloader: Unexpected state transition attempted, aborted: 2->1 (->)
```
After that the library is maybe in an invalid state, because the second attempt just freezes:
```
mosh:~$ fota download http://test-eu.nrfcloud.com/small_1024.txt foo
FOTA: Starting download...
```
The timeout could be caused by the library trying to use TLS. The protocol in URI is HTTP, but a sec_tag is provided. Anyway, at least the library should not end up in a state where it no longer works.

Reply:
Thanks for testing this. I've been able to reproduce it and changed the behavior to not use secure connection when `http://` is in the URI and a certificate is given. 

Before this I updated the fota library and downloader states behavior so that the second attempt does not hang.